setup.cfg

```diff
@@ -51,35 +51,40 @@ keywords =
 
 [options]
 zip_safe = True
-install_requires = 
+install_requires =
+    appdirs ~= 1.4
 	depinfo
+    diskcache ~= 5.0
 	future
+    httpx ~= 0.14
 	importlib_resources
 	numpy ~= 1.13
 	optlang ~= 1.4
 	pandas ~= 1.0
+    pydantic ~= 1.6
```

Comment:
Do you need to trigger validation with pydantic or does that happen automatically during instantiation? I don't think it adds that much to the codebase here but if you have plans to use it in more places it may be okay.

Reply:
Validation happens at instantiation time indeed. It can also be configured to additionally run at assignment. It does not add incredibly much for the moment but I would love to use it for our `dict`, `json`, `yaml` IO as one can automatically generate a JSON schema from pydantic models.

Reply:
How fast is pydantic? Model loading needs to be pretty performant so that may generate some overhead. (Sorry, this is a bit off-topic).

Reply:
It's the most performant of the structured data packages that I know. Can be further improved by using something like `ujson` or `orjson`. Should run some benchmarks first, of course, but I think it's a huge advantage to be able to define the data structures as classes, have good validation, and be able to generate a JSON schema automatically.